pkg/kubelet/dockertools/manager.go

```diff
@@ -2058,13 +2063,25 @@ func getPidMode(pod *api.Pod) string {
 func getIPCMode(pod *api.Pod) string {
 	ipcMode := ""
 	if pod.Spec.SecurityContext != nil && pod.Spec.SecurityContext.HostIPC {
-		ipcMode = "host"
+		ipcMode = namespaceModeHost
 	}
 	return ipcMode
 }
 
-// GetNetNs returns the network namespace path for the given container
-func (dm *DockerManager) GetNetNs(containerID kubecontainer.ContainerID) (string, error) {
+// IsUsingHostNetwork returns true if the provided container ID is using host
+// networking.  This method is used by network plugins to determine if a pod
+// requires additional setup. Pods running with host networking do not.
+func (dm *DockerManager) IsUsingHostNetwork(containerID kubecontainer.ContainerID) (bool, error) {
```

Comment:
I agreed that we don't have a better way for now, but disagree the argument of that this isolates between kubelet and net plugin. By design, container runtime should understand pod, and network namespace is setup for pod anyway. In this case, retrieve this information from Spec itself is way better than inspecting from docker container.